lib/phoenix/live_dashboard/page_builder.ex

```diff
@@ -795,6 +792,63 @@ defmodule Phoenix.LiveDashboard.PageBuilder do
     """
   end
 
+  @doc false
+  attr :id, :string,
+    required: true,
+    doc: "Because is a stateful `Phoenix.LiveComponent` an unique id is needed."
+
+  attr :data, :list,
+    default: [],
+    doc: """
+    Temporary list of points to show in the chart.
+    Each element should be the format `{optional_label, x, y}`.
+    New points can be added using the function `send_data_to_chart/2` in real time.
+    """
+
+  attr :title, :string, required: true, doc: "Title of the chart"
+  attr :hint, :string, default: nil, doc: "A textual hint to show close to the title."
+
+  attr :kind, :atom,
+    values: [:counter, :last_value, :sum, :summary, :distribution],
+    doc: "Kind of chart to use."
+
+  attr :label, :string, default: nil, doc: "Default label to use in the chart."
+  attr :tags, :list, default: [], doc: "Optional list of tags."
+  attr :prune_threshold, :integer, default: 1_000, doc: "Number of points to keep before pruning."
+  attr :unit, :string, default: "", doc: "The unit that represent the chart."
+
+  attr :bucket_size, :integer,
+    doc: "Bucket size for histogram. Default: 20 when `kind = :histogram`, otherwise `nil`."
+
+  attr :full_width, :boolean, default: false, doc: "Size of the chart"
+
+  def live_chart(assigns) do
+    assigns =
+      assign_new(assigns, :bucket_size, fn ->
+        if assigns.kind == :histogram, do: 20, else: nil
+      end)
+
+    ~H"""
+    <.live_component
+      module={ChartComponent}
+      id={@id}
+      title={@title}
+      hint={@hint}
+      kind={@kind}
+      label={@label}
+      tags={@tags}
+      prune_threshold={@prune_threshold}
+      unit={@unit}
+      bucket_size={@bucket_size}
+      full_width={@full_width}
+    />
+    """
+  end
+
+  def send_data_to_chart(id, data) do
+    Phoenix.LiveView.send_update(ChartComponent, id: id, data: data)
+  end
```

Comment:
We would need to `@doc false` this too but I am thinking if we can just move it to `ChartComponent` and remove the component above? We only call the component above once and we build the parameters dynamically, so it is not like we are benefiting from `attr`as well. Thoughts?

Reply:
We use `live_chart/1` and `send_data_to_chart/2` on the metrics page, but now on the memory allocators page too! For this reason, I did the whole refactor. We cannot send a "metric" as previously worked the chart component to the memory allocator chart. I'll keep it here for the moment. If you really want to change it, leave me a comment and I'll do it in the following PR :)

Reply:
I see! So we can keep `live_chart` as is but I would move `send_data_to_chart` to ChartComponent.send_data. The only reason is that those functions will be imported into the user code, and we want to hide them for now.

If you really want to keep it on `page_builder`, then we should `@doc false` it. :)